plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -420,20 +422,60 @@ int hash_rules(const std::vector<std::string> &irbs, const int pid, std::string
 
 class make_copy {
 public:
-        make_copy(const std::vector<std::string>& _irbs, const int _pid) : irbs_(_irbs), pid_(_pid) {
-            for(auto const &irb : _irbs) {
-                boost::filesystem::copy_file(get_rule_base_path(irb), get_rule_base_path_copy(irb, _pid));
-            }
+  make_copy(const std::vector<std::string>& _irbs)
+      : irbs_(_irbs)
+  {
+      for (auto const& irb : _irbs) {
+          std::string rulebase;
+          std::ifstream ifs(get_rule_base_path(irb));
+          rule_bases.insert(
+              {irb, std::string({std::istreambuf_iterator<char>(ifs), std::istreambuf_iterator<char>()})});
+      }
+  }
+
+  std::string& get_rulebase(const std::string& irb)
```

Comment:
Are the strings returned by this function meant to be mutable?

Reply:
No, but originally marking it `const` caused issues with trying to pass it into `fmemopen`, since it expects a non-const input pointer, even if I'm opening it read-only.
I was originally worried I might hit a weird UB case if I took address-of a `const` variable, but some research says it's technically not illegal, so I could do it and maybe just `const_cast` away the const for `fmemopen`. Would that be preferable?

Reply:
No. I believe casting away const on a const object is UB. A comment explaining why the string is a mutable reference is good enough.

Reply:
If I recall correctly, we've previously allowed it because it isn't UB unless you access it as if it were non-const. Does that change your mind about the outcome?